src/amo/components/Collection/index.js

```diff
@@ -247,15 +249,36 @@ export class CollectionBase extends React.Component<InternalProps> {
     invariant(slug, 'slug is required');
     invariant(username, 'username is required');
 
+    const { pageSize } = collection;
+    const pages = pageSize ? (collection.numberOfAddons - 1) / pageSize : null;
+    const isNewPage = pages && pages % 1 === 0;
+    const newFilter = (pages && Math.ceil(pages)) || 1;
```

Comment:
3 lines of basic maths, yet I don't understand anything..

I would like to see a helper function above the class with a name explaining what it is supposed to do, and we should not mix `&&` and `||` and ternary operators. I would like to avoid the modulo 1 as well, because this is JS black magic.

Reply:
yikes ok : )let me see if I can clarify and clean up a bit